cpp/src/arrow/ipc/json-internal.cc

```diff
@@ -940,6 +954,108 @@ class JsonArrayReader {
   }
 
   template <typename T>
+  typename std::enable_if<std::is_base_of<TimestampType, T>::value, Status>::type ReadArray(
+      const RjObject& json_array, int32_t length, const std::vector<bool>& is_valid,
+      const std::shared_ptr<DataType>& type, std::shared_ptr<Array>* array) {
+
+    const auto& json_data = json_array.FindMember("DATA");
+    RETURN_NOT_ARRAY("DATA", json_data, json_array);
+
+    const auto& json_data_arr = json_data->value.GetArray();
+    DCHECK_EQ(static_cast<int32_t>(json_data_arr.Size()), length);
+
+    typename TypeTraits<T>::BuilderType builder(pool_, type);
+    for (int i = 0; i < length; ++i) {
+      if (!is_valid[i]) {
+        builder.AppendNull();
+        continue;
+      }
+
+      const rj::Value& val = json_data_arr[i];
+      // TODO: Can we have signed ints in Timestamp?
+      if (IsSignedInt<T>::value) {
+        DCHECK(val.IsInt() || val.IsInt64());
+        builder.Append(val.GetInt64());
```

Comment:
Good to know. Suppose we'll have the proof in the working integration tests anyway